### ISSUE TITLE
Pool.php wrong anotation for promise()

### ISSUE THREAD

**Issue by lukasojd** [guzzle/guzzle#2469]
**Guzzle version(s) affected**: 6.5.0 

**Description**  
https://github.com/guzzle/guzzle/blob/6.5/src/Pool.php#L74

Function promomise() has wrong return annotation.

Return typehint have anotation GuzzleHttp\Promise\Promise  and class has namespace GuzzleHttp.  Return type would be return \GuzzleHttp\Promise\Promise or Promise\Promise...

Right: 
```
    /**
     * Get promise
     * @return Promise\Promise
     */
    public function promise()
    {
        return $this->each->promise();
    }

```
or 
```
    /**
     * Get promise
     * @return \GuzzleHttp\Promise\Promise
     */
    public function promise()
    {
        return $this->each->promise();
    }
```

**Comment by alexeyshockov** [comment]
It's already fixed in `master` with a type hint (for PHP 7+) and will be released as a part of Guzzle 7.0. If you want this fix for `6.5`, please make a PR to `6.5` branch.

**Comment by lukasojd** [comment]
Task is duplicate https://github.com/guzzle/guzzle/pull/2467. I am closing task
